### ISSUE TITLE
Bug in no-unused-vars when using extra parens to avoid no-cond-assign

### ISSUE THREAD

**Issue by bedney** [eslint/eslint#1918]
Sorry if this has been logged before. There are a quite a few logged against `no-unused-vars`, but I didn't see this one.

I use additional parens to avoid `no-cond-assign`:

```js
function foo(aThing) {

    var something;

    if ((something = aThing)) {
        console.log('got something');
    }
}

foo('a thing');
```

In this case, `something` is reported as an unused var, even though it is obviously being used.

Changing the code to this causes the lint error to go away:

```js
function foo(aThing) {

    var something;

    something = aThing;
    if (something) {
        console.log('got something');
    }
}

foo('a thing');
```

**Comment by nzakas** [comment]
The rule is behaving as designed. In your example, you are only assigning to the variable and never using that value anywhere. To put simply: you must read a variable for it to be considered "used".

**Comment by bedney** [comment]
You are absolutely correct. Sorry for the erroneous report.
